samples/matter/lock/src/credentials/credentials_data_types.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "credentials_data_types.h"
+
+namespace DoorLockData
+{
+
+void pack(void *buff, const void *data, size_t dataSize, size_t &offset)
+{
+	memcpy(reinterpret_cast<uint8_t *>(buff) + offset, data, dataSize);
+	offset += dataSize;
+}
+
+void unpack(void *data, size_t dataSize, const void *buff, size_t &offset)
+{
+	memcpy(data, reinterpret_cast<const uint8_t *>(buff) + offset, dataSize);
+	offset += dataSize;
+}
+
+CHIP_ERROR Credential::FillFromPlugin(EmberAfPluginDoorLockCredentialInfo &credentialInfo)
+{
+	if (credentialInfo.credentialData.size() < kMaxCredentialLength) {
+		mInfo.mFields.mStatus = static_cast<uint8_t>(credentialInfo.status);
```

Comment:
Is there a reason we don't want to use Matter enums and cast things to `uint8_t`?

Reply:
I guess we wanted to make the structure more suitable for the serialization, but in fact all underlying enums are inheriting from `uint8_t` anyway and also pack/unpack accept `void*`, so I believe this could be refactored. I hope it's fine to do it as a follow up.

Reply:
Task created to refactor credentials_data_types.